curiosrc/seal/task_treed.go

```diff
@@ -23,16 +22,62 @@ import (
 	"github.com/filecoin-project/lotus/storage/sealer/storiface"
 )
 
-type TreesTask struct {
+type TreeDTask struct {
 	sp *SealPoller
 	db *harmonydb.DB
 	sc *ffi.SealCalls
 
 	max int
 }
 
-func NewTreesTask(sp *SealPoller, db *harmonydb.DB, sc *ffi.SealCalls, maxTrees int) *TreesTask {
-	return &TreesTask{
+func (t *TreeDTask) CanAccept(ids []harmonytask.TaskID, engine *harmonytask.TaskEngine) (*harmonytask.TaskID, error) {
+	if engine.Resources().Gpu > 0 {
+		return &ids[0], nil
+	}
+	return nil, nil
+}
+
+func (t *TreeDTask) TypeDetails() harmonytask.TaskTypeDetails {
+	ssize := abi.SectorSize(32 << 30) // todo task details needs taskID to get correct sector size
+	if isDevnet {
+		ssize = abi.SectorSize(2 << 20)
+	}
+
+	return harmonytask.TaskTypeDetails{
+		Max:  t.max,
+		Name: "SDRTreeD",
+		Cost: resources.Resources{
+			Cpu:     1,
+			Ram:     64 << 20, // todo
+			Gpu:     0,
+			Storage: t.sc.Storage(t.taskToSector, storiface.FTSealed, storiface.FTCache, ssize, storiface.PathSealing, 1),
```

Comment:
The sealed file is created in TreeR
```suggestion
			Storage: t.sc.Storage(t.taskToSector, storiface.FTNone, storiface.FTCache, ssize, storiface.PathSealing, 1),
```